src/jdk.compiler/share/classes/com/sun/tools/javac/launcher/MemoryClassLoader.java

```diff
@@ -271,6 +270,22 @@ public URL nextElement() {
         };
     }
 
+    /**
+     * Resolves a "resource name" (as used in the getResource* methods)
+     * to an existing file relative to source root path, or null otherwise.
+     *
+     * @param name the resource name
+     * @return the URL of the resource, or null
+     */
+    private URL toResourceInRootPath(String name) {
+        try {
+            var path = Resources.toFilePath(programDescriptor.sourceRootPath(), name);
+            return path == null ? null : path.toUri().toURL();
+        } catch (IOException e) {
```

Comment:
toUri throws IOError so I think you will need handle this too, otherwise I think the updated toResourceInRootPath looks okay.

Reply:
Caught and handled in https://github.com/openjdk/jdk/pull/20193/commits/6a0db8a6c7e630e37a431463b67749303a6e3d13